docs/core/testing/unit-testing-mstest-migration-from-v1-to-v3.md

```diff
@@ -0,0 +1,290 @@
+
+- **Future-Proofing Your Tests**: MSTest v3 offers robust support for modern .NET versions, including .NET 6, .NET 7, and future iterations, along with cross-platform compatibility. This means your tests will be more adaptable and ready for upgrades, avoiding the technical debt of outdated testing frameworks.
+
+- **Increased Code Reliability with New Analyzers**: MSTest v3 built-in code analyzers enforce best practices, helping to catch issues early and promoting cleaner, more maintainable test code:
+  - **Proactive Issue Detection**: The analyzers provide real-time feedback, suggesting improvements and flagging potential issues as you write your tests.
+  - **Stronger Type-Safe Assertions**: Replacing ambiguous overloads, MSTest v3 enforces type-safe assertions, reducing the risk of incorrect test behavior and making tests more reliable.
+  - **Cleaner Code, Less Maintenance**: By aligning your tests with MSTest standards and best practices, MSTest v3 helps you maintain a cleaner, more manageable codebase, reducing technical debt over time.
+
+- **Greater Flexibility and Extensibility**: MSTest v3 supports advanced testing scenarios, including dynamic data sources and in-assembly parallel execution. This flexibility enables more sophisticated testing approaches and speeds up test suites without complex configurations.
+
+By upgrading, you’re setting up your tests to be faster, more reliable, and adaptable to future .NET developments, positioning your project for long-term success and easier maintenance.
+
+---
+
+## Steps to Migrate
+
+### 1. Remove Assembly Reference
+
+For projects using MSTest v1 through assembly references, it’s common to find references to either or both of the following DLLs:
+
+- `Microsoft.VisualStudio.QualityTools.UnitTestFramework`
+- `Microsoft.VisualStudio.TestPlatform.TestFramework`
+
+In non-SDK style projects, these references are often added through Visual Studio rather than by directly editing the XML. To remove these references using the Visual Studio GUI:
+
+1. **Open Solution Explorer** in Visual Studio.
+2. **Right-click on the project** using MSTest and select **Properties**.
+3. Navigate to the **References** tab.
+4. Locate and **select the MSTest DLL references** (`Microsoft.VisualStudio.QualityTools.UnitTestFramework` and/or `Microsoft.VisualStudio.TestPlatform.TestFramework`).
+5. **Right-click** the selected reference(s) and choose **Remove**.
+
+For those comfortable editing XML directly, you can manually remove the references from your `.csproj` file as follows:
+
+```xml
+<Reference Include="Microsoft.VisualStudio.QualityTools.UnitTestFramework, Version=10.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+    <Private>False</Private>
+</Reference>
+```
+
+### 2. Update Your Project
```

Comment:
This whole section is forgetting about MSTest.TestAdapter

Reply:
Yes, we have that now, but old projects that are using mstest v1, are using MSTest.TestAdapter and MSTest.TestFramework nuget packages, and most likely are using non-sdk projects. It it recommended to upgrade to MSTest meta-package, but you are not describing anything about the nugets they are using. 

So at best they will end up with all three of them installed directly in the project.

It would be better if you avised them to update MSTest.TestFramework and MSTest.TestAdapter packages to latest, OR that they considered removing them and installed MSTest package instead, that will bring both of them.